components/pdf_preview.jsx

```diff
@@ -111,19 +133,35 @@ export default class PDFPreview extends React.PureComponent {
     }
 
     onDocumentLoad = (pdf) => {
-        const numPages = pdf.numPages <= MAX_PDF_PAGES ? pdf.numPages : MAX_PDF_PAGES;
-        this.setState({pdf, numPages});
-        for (let i = 1; i <= pdf.numPages; i++) {
-            pdf.getPage(i).then(this.onPageLoad);
+        this.setState({pdf, numPages: pdf.numPages});
+        for (let i = 0; i < pdf.numPages; i++) {
+            this[`pdfCanvasRef-${i}`] = React.createRef();
         }
+        this.loadPages(pdf);
     }
 
     onDocumentLoadError = (reason) => {
         console.log('Unable to load PDF preview: ' + reason); //eslint-disable-line no-console
         this.setState({loading: false, success: false});
     }
 
-    onPageLoad = (page) => {
+    loadPages = async (pdf) => {
+        const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+        for (let i = 0; i < pdf.numPages; i++) {
+            await this.loadPage(pdf, i); // eslint-disable-line no-await-in-loop
+
+            // Give time to render the interface
+            await sleep(10); // eslint-disable-line no-await-in-loop
```

Comment:
If i don't do this, this takes a long time to render for big documents because it consumes all the cpu time loading the pdf files and is not able to render.